test/e2e/autotls/domain_mapping_test.go

```diff
@@ -140,3 +155,145 @@ func TestDomainMappingAutoTLS(t *testing.T) {
 	httpsClient := createHTTPSClient(t, clients, svc, rootCAs)
 	ingress.RuntimeRequest(ctx, t, httpsClient, "https://"+host)
 }
+
+func TestBYOCertificate(t *testing.T) {
+	if !test.ServingFlags.EnableAlphaFeatures {
+		t.Skip("Alpha features not enabled")
+	}
+	clients := test.Setup(t)
+
+	names := test.ResourceNames{
+		Service: test.ObjectNameForTest(t),
+		Image:   test.PizzaPlanet1,
+	}
+
+	test.EnsureTearDown(t, clients, &names)
+
+	// service to be mapped
+	ksvc, err := v1test.CreateServiceReady(t, clients, &names)
+	if err != nil {
+		t.Fatalf("Failed to create initial Service %v: %v", names.Service, err)
+	}
+
+	host := ksvc.Service.Name + ".example.org"
+	// set resolvabledomain for custom domain to false by default.
+	resolvableCustomDomain := false
+
+	if test.ServingFlags.CustomDomain != "" {
+		host = ksvc.Service.Name + "." + test.ServingFlags.CustomDomain
+		resolvableCustomDomain = true
+	}
+
+	cert, key := makeCertificateForDomain(t, host)
+	secret, err := clients.KubeClient.CoreV1().Secrets(test.ServingNamespace).Create(context.Background(), &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: test.AppendRandomString("imasecret"),
+		},
+		Type: corev1.SecretTypeTLS,
+		Data: map[string][]byte{
+			corev1.TLSCertKey:       cert,
+			corev1.TLSPrivateKeyKey: key,
+		},
+	}, metav1.CreateOptions{})
+
+	if err != nil {
+		t.Fatalf("Domainmapping creation could not be completed: %v", err)
+	}
+
+	t.Cleanup(func() {
+		err = clients.KubeClient.CoreV1().Secrets(secret.Namespace).Delete(context.Background(), secret.Name, metav1.DeleteOptions{})
+		if err != nil {
+			t.Fatalf("Secret %s/%s could not be deleted: %v", secret.Namespace, secret.Name, err)
+		}
+	})
+
+	dm := v1alpha1.DomainMapping{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      host,
+			Namespace: ksvc.Service.Namespace,
+		},
+		Spec: v1alpha1.DomainMappingSpec{
+			Ref: duckv1.KReference{
+				APIVersion: "serving.knative.dev/v1",
+				Name:       ksvc.Service.Name,
+				Namespace:  ksvc.Service.Namespace,
+				Kind:       "Service",
+			}, TLS: &v1alpha1.SecretTLS{
+				SecretName:      secret.Name,
+				SecretNamespace: secret.Namespace,
+			}},
+		Status: v1alpha1.DomainMappingStatus{},
+	}
+
+	t.Log("HOST===>", host)
+	_, err = clients.ServingAlphaClient.DomainMappings.Create(context.Background(), &dm, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("Problem creating domainmapping : %v", err)
```

Comment:
Let's print the host here

```suggestion
		t.Fatalf("Problem creating DomainMapping %q: %v", host, err)
```